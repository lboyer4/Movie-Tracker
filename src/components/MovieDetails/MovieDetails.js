import React from 'react';
import './_MovieDetails.scss';
import PropTypes from 'prop-types';

const MovieDetails = ({ movie_id, title, poster_path, release_date, vote_average, overview, favorited}) => {
	return (
		<article>
			<div className="left">
				<img src={poster_path} alt='movie poster'/>
			</div>
			<div className="right">
				<h1 className='movie-title-detail'>{title}</h1>
				<h5 className='date-detail'>Release Date: {release_date}</h5>
				<p className='overview-details'> {overview} </p>
				<h6 className='vote-detail'> Vote Average: {vote_average}</h6>
			</div>
		</article>
	)
}

export default MovieDetails;

MovieDetails.propTypes = {
	favorited: PropTypes.bool,
	movie_id: PropTypes.number,
	overview: PropTypes.string,
	poster_path: PropTypes.string,
	release_date: PropTypes.string,
	title: PropTypes.string,
	vote_average: PropTypes.number
}