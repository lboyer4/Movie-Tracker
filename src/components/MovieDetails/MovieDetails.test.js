import React from 'react';
import { shallow } from 'enzyme';
import MovieDetails from './MovieDetails';
import { mockSelectedMovie } from '../../utils/mockData';

describe('MovieDetails', () => {
	it('should match snapshot without a selected movie', () => {
		const wrapper = shallow(
			<MovieDetails />
		)
		expect(wrapper).toMatchSnapshot()
	});

	it('should match snapshot with a selected move', () => {
		const wrapper = shallow(
			<MovieDetails
				selectedMovie={ mockSelectedMovie }
			/>
		);
	});
});