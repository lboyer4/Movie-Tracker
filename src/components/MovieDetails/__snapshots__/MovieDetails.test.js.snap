// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails should match snapshot without a selected movie 1`] = `
<article>
  <div
    className="left"
  >
    <img
      alt="movie poster"
    />
  </div>
  <div
    className="right"
  >
    <h1
      className="movie-title-detail"
    />
    <h5
      className="date-detail"
    >
      Release Date: 
    </h5>
    <p
      className="overview-details"
    >
       
       
    </p>
    <h6
      className="vote-detail"
    >
       Vote Average: 
    </h6>
  </div>
</article>
`;
